### PR TITLE
PWGHF: Update QA tasks for vertexing and MC PID

### PR DESCRIPTION
**Pull request by njacazio**
- Extend vertexing QA
- Add TOF PID task for MC
 - Fill PID info for different PDG codes
 - Separating info for primaries and secondaries

### CODE REVIEW

**Comment by njacazio** [comment]
This PR fixes also the issue of the units in the poiniting resolution

File: Analysis/Tasks/pidTOFqa.cxx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+               aod::McParticles& mcParticles)
+  {
+    const float collisionTime_ps = collision.collisionTime() * 1000.f;
+    for (auto t : tracks) {
+      //
+      if (t.tofSignal() < 0) { // Skipping tracks without TOF
+        continue;
+      }
+      float nsigma = -999.f;
+      if (pid_type == 0) {
```
Comment (jgrosseo) [inline Analysis/Tasks/pidTOFqa.cxx]: Doesn't this have to be constexpr?

Comment (njacazio) [reply Analysis/Tasks/pidTOFqa.cxx]: Indeed, however it worked without. I fixed it in the new version

Comment (jgrosseo) [reply Analysis/Tasks/pidTOFqa.cxx]: Sure, but it is done run time and therefore much slower...

**Review by jgrosseo — DISMISSED** [review]

File: Analysis/Tasks/PWGHF/qaTaskEfficiency.cxx [hunk 2]
```diff
@@ -71,17 +71,19 @@ struct QATrackingEfficiencyPt {
                              phiMin.value, phiMax.value,
                              selPrim.value);
     const TString xaxis = "#it{p}_{T} GeV/#it{c}";
+    o2fw::AxisSpec axis_pt{ptBins.value, ptMin.value, ptMax.value};
 
     histos.add("num", "Numerator " + tag + ";" + xaxis,
-               o2fw::kTH1D, {{ptBins.value, ptMin.value, ptMax.value}});
+               o2fw::kTH1D, {axis_pt});
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/qaTaskEfficiency.cxx]: A minor suggestion:
`axis_pt` -> `axisPt`
`v_reco` -> `vecReco`

**Review by vkucera — CHANGES_REQUESTED** [review]

**Review by ginnocen — DISMISSED** [review]
hi @njacazio this is fine for me, please just implement Vit's suggestion.

**Comment by njacazio** [comment]
I took advantage of the additional request to include the efficiency vs eta and phi

**Review by ginnocen — APPROVED** [review]

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**